hedera-node/hedera-consensus-service-impl/src/main/java/com/hedera/node/app/service/consensus/impl/handlers/ConsensusDeleteTopicHandler.java

```diff
@@ -82,14 +85,9 @@ public void handle(@NonNull final HandleContext context) {
 
         final var op = context.body().consensusDeleteTopicOrThrow();
         final var topicStore = context.writableStore(WritableTopicStore.class);
-        var topicId = op.topicIDOrElse(TopicID.DEFAULT);
-        var optionalTopic = topicStore.get(topicId);
-
-        /* If the topic doesn't exist, return INVALID_TOPIC_ID */
-        if (optionalTopic.isEmpty()) {
-            throw new HandleException(INVALID_TOPIC_ID);
-        }
-        final var topic = optionalTopic.get();
+        final var topicId = op.topicIDOrElse(TopicID.DEFAULT);
+        // preHandle already checks for topic existence
+        final Topic topic = topicStore.getTopic(topicId);
```

Comment:
We should still validate topic is not null right ?

Reply:
We checked topic is not null in preHandle() from ReadableTopicStore. If code get here in handle(), preHandle() has to be success, do we still need to check if topic is null in WritableTopicStore?

Reply:
yes that makes sense

Reply:
Data in store can change between `preHandle` and `Handle`.
We cannot assume any state validated in `preHandle` remains valid in `Handle`.

Reply:
`preHandle` is called again right before `handle` with the state that will be used in `handle`

Reply:
In `HandleWorkflow.handleUserTransaction()`, we call `preHandleWorkflow.preHandleTransaction() `again before `dispatchHandle()`.